generator/internal/stub_factory_generator.cc

```diff
@@ -95,24 +98,41 @@ Status StubFactoryGenerator::GenerateCc() {
 std::shared_ptr<$stub_class_name$>
 CreateDefault$stub_class_name$(
     google::cloud::CompletionQueue cq, Options const& options) {
-  auto auth = google::cloud::internal::CreateAuthenticationStrategy(
-      std::move(cq), options);
-  auto channel = auth->CreateChannel(
-    options.get<EndpointOption>(), internal::MakeChannelArguments(options));
-  auto service_grpc_stub = $grpc_stub_fqn$::NewStub(channel);)""");
+  auto endpoint_option = [&]() -> absl::optional<std::string> {
+    if (!options.has<EndpointOption>()) return absl::nullopt;
+    return options.get<EndpointOption>();
+  }();
```

Comment:
I presume something like this will need to appear in the hand-written libraries too, so perhaps we should add an API like [this](https://github.com/googleapis/google-cloud-cpp/blob/75261bbabe3aea4f7b5eb6b0a8a850cc47d89fa9/google/cloud/spanner/client.cc#L38-L45) to `google/cloud/options.h` first.